src/routes/local.ts

```diff
@@ -1,39 +1,97 @@
-import express from 'express'
-import n2yo from './../api/n2yo/index.js'
-import SatelliteUtilities from '../utils/satelliteUtils'
-import _ from 'lodash'
-import SatelliteInterface from '../utils/types/satelliteInterface.js'
+import express from "express";
+import n2yo from "./../api/n2yo/index.js";
+import SatelliteUtilities from "../utils/satelliteUtils";
+import _ from "lodash";
+import SatelliteInterface from "../utils/types/satelliteInterface.js";
+import { object, number, string, assert } from "superstruct";
 
-const router = express.Router()
-const PROPS_TO_SEND_IN_RESPOSNE = ['_id', 'officialName', 'perigee', 'apogee', 'operator', 'operatorCountry', 'purpose']
+const router = express.Router();
+const PROPS_TO_SEND_IN_RESPOSNE = [
+  "_id",
+  "officialName",
+  "perigee",
+  "apogee",
+  "operator",
+  "operatorCountry",
+  "purpose",
+];
 
-router.get('/two-line-elements/:id', (req, res) => {
-    if (req.params.id === undefined) return res.sendStatus(400)
-    n2yo.getTwoLineElements(req.params.id)
-        .then(result => res.json(result))
-        .catch(() => res.sendStatus(400))
-})
+router.get("/two-line-elements/:id", (req, res) => {
+  if (req.params.id === undefined) return res.sendStatus(400);
+  n2yo
+    .getTwoLineElements(req.params.id)
+    .then((result) => res.json(result))
+    .catch(() => res.sendStatus(400));
+});
 
-router.get('/satellite', (req, res) => {    
-    if (typeof req.query.id === 'string') {
-        return SatelliteUtilities.getSatelliteById(req.query.id)
-            .then((satelliteDoc) => {
-                if (!satelliteDoc) return res.sendStatus(400)
-                return res.json(_.pick(satelliteDoc, PROPS_TO_SEND_IN_RESPOSNE))
-            })
-            .catch(() => { return res.sendStatus(400) })
-    }
-    if (typeof req.query.name === 'string') {
-        return SatelliteUtilities.getSatelliteByName(req.query.name)
-            .then((satelliteDoc) => {
-                if (!satelliteDoc) return res.sendStatus(400)
-                const satellitesArray = <Partial<SatelliteInterface>[]>
-                    satelliteDoc.map((satellite) => _.pick(satellite, PROPS_TO_SEND_IN_RESPOSNE))
-                return res.json(satellitesArray)
-            })
-            .catch(() => { return res.sendStatus(400) })
-    }
-    res.sendStatus(400)
-})
+router.get("/satellite", (req, res) => {
```

Comment:
This route works with "id" or "name" query parameters, but not with both at the same time. I suggest splitting this route into two routes: `/satellite/by-id/:id` and `/satellite/by-name/:name`